supertokens_python/session/session_recipe.py

```diff
@@ -0,0 +1,247 @@
+    def init(config=None):
+        def func(app_info: AppInfo):
+            if SessionRecipe.__instance is None:
+                SessionRecipe.__instance = SessionRecipe(SessionRecipe.recipe_id, app_info, config)
+                return SessionRecipe.__instance
+            else:
+                raise_general_exception(None, 'Session recipe has already been initialised. Please check your code for bugs.')
+        return func
+
+    @staticmethod
+    def get_instance() -> SessionRecipe:
+        if SessionRecipe.__instance is not None:
+            return SessionRecipe.__instance
+        raise_general_exception(None, 'Initialisation not done. Did you forget to call the SuperTokens.init function?')
+
+    @staticmethod
+    def reset():
+        if ('SUPERTOKENS_ENV' not in environ) or (
+                environ['SUPERTOKENS_ENV'] != 'testing'):
+            raise_general_exception(None, 'calling testing function in non testing env')
+        SessionRecipe.__instance = None
+
+    # instance functions below...............
+
+    async def get_handshake_info(self) -> HandshakeInfo:
+        if self.handshake_info is None:
+            ProcessState.get_instance().add_state(AllowedProcessStates.CALLING_SERVICE_IN_GET_HANDSHAKE_INFO)
+            response = await self.get_querier().send_post_request(NormalisedURLPath(self, '/recipe/handshake'), {})
+            self.handshake_info = HandshakeInfo({
+                **response,
+                'antiCsrf': self.config.anti_csrf
+            })
+        return self.handshake_info
+
+    def update_jwt_signing_public_key_info(self, new_key, new_expiry):
+        if self.handshake_info is not None:
+            self.handshake_info.update_jwt_signing_public_key_info(new_key, new_expiry)
+
+    async def create_new_session(self, request: BaseRequest, user_id: str, jwt_payload: Union[dict, None] = None,
+                                 session_data: Union[dict, None] = None) -> Session:
+        session = await session_functions.create_new_session(self, user_id, jwt_payload, session_data)
+        access_token = session['accessToken']
+        refresh_token = session['refreshToken']
+        id_refresh_token = session['idRefreshToken']
+        new_session = Session(self, access_token['token'], session['session']['handle'],
+                                            session['session']['userId'], session['session']['userDataInJWT'])
+        new_session.new_access_token_info = access_token
+        new_session.new_refresh_token_info = refresh_token
+        new_session.new_id_refresh_token_info = id_refresh_token
+        if 'antiCsrfToken' in session and session['antiCsrfToken'] is not None:
+            new_session.new_anti_csrf_token = session['antiCsrfToken']
+        request.set_session(new_session)
+        return request.get_session()
+
+    async def get_session(self, request: BaseRequest, anti_csrf_check: Union[bool, None] = None, session_required: bool = True) -> Union[Session, None]:
```

Comment:
this is a use facing function which means it can be directly used by the user in their API handler. We shouldn't expect them to wrap the request based onto their framework. Rather check here if the request is wrapped or not. if it is not wrapped, then wrap it based on the framework config